google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/TestEnvRule.java

```diff
@@ -79,6 +79,9 @@ protected void before() throws Throwable {
   protected void after() {
     try {
       env().cleanUpStale();
+      if ("prod".equals(env) || "direct_path".equals(env)) {
+        env().cleanUpInstances();
+      }
```

Comment:
I think it would be better to keep this logic local to AbstractTestEnv subclasses.
Maybe add a getter isInstanceAdminSupported() to AbstractTestEnv and default it to true. Then have EmulatorEnv override it with false.

Reply:
Thanks for the suggestion, have updated with the same approach. Please have a look.